src/numberparse.rs

```diff
@@ -133,26 +129,14 @@ pub enum Number {
 }
 
 #[cfg_attr(not(feature = "no-inline"), inline)]
+//#[cfg(target_arch = "aarch64")]
 fn parse_eight_digits_unrolled(chars: &[u8]) -> u32 {
-    unsafe {
-        // this actually computes *16* values so we are being wasteful.
-        let ascii0: __m128i = _mm_set1_epi8(b'0' as i8);
-        let mul_1_10: __m128i =
-            _mm_setr_epi8(10, 1, 10, 1, 10, 1, 10, 1, 10, 1, 10, 1, 10, 1, 10, 1);
-        let mul_1_100: __m128i = _mm_setr_epi16(100, 1, 100, 1, 100, 1, 100, 1);
-        let mul_1_10000: __m128i = _mm_setr_epi16(10000, 1, 10000, 1, 10000, 1, 10000, 1);
-        // We know what we're doing right? :P
-        #[allow(clippy::cast_ptr_alignment)]
-        let input: __m128i = _mm_sub_epi8(
-            _mm_loadu_si128(chars.get_unchecked(0..16).as_ptr() as *const __m128i),
-            ascii0,
-        );
-        let t1: __m128i = _mm_maddubs_epi16(input, mul_1_10);
-        let t2: __m128i = _mm_madd_epi16(t1, mul_1_100);
-        let t3: __m128i = _mm_packus_epi32(t2, t2);
-        let t4: __m128i = _mm_madd_epi16(t3, mul_1_10000);
-        _mm_cvtsi128_si32(t4) as u32 // only captures the sum of the first 8 digits, drop the rest
-    }
+    let val: u64 = unsafe { *(chars.as_ptr() as *const u64) };
```

Comment:
we shouldn't fall back to the general "slow" implementation on all architectures, lets make this architecture dependant and keep the x86 version SIMD'd and the ARM version plain.